library/common/extensions/filters/http/platform_bridge/filter.cc

```diff
@@ -21,18 +21,47 @@ PlatformBridgeFilterConfig::PlatformBridgeFilterConfig(
           Api::External::retrieveApi(proto_config.platform_filter_name()))) {}
 
 PlatformBridgeFilter::PlatformBridgeFilter(PlatformBridgeFilterConfigSharedPtr config)
-    : platform_filter_(config->platform_filter()) {}
+    : platform_filter_(*config->platform_filter()) {
+  // The initialization above sets platform_filter_ to a copy of the struct stored on the config.
+  // In the typical case, this will represent a filter implementation that needs to be intantiated.
+  // static_context will contain the necessary platform-specific mechanism to produce a filter
+  // instance. instance_context will initially be null, but after initialization, set to the
+  // context needed for actual filter invocations.
+
+  // We treat a null initializer as a static implementation.
```

Comment:
Is this a feature we want to support? i.e do we want it to be possible to have static filters? While potentially useful in some cases I worry that giving options might make the entire affair a little more confusing.

Reply:
I'm primarily interested in supporting it right now because I like the convention of being able to `calloc` these structs as a safety convention. I wasn't planning on exposing it above this level at this point.

I agree about not introducing extra complexity, but I also realized it didn't take very much to support it generally here. Assuming we move forward, I will ensure a test covers this case.

Reply:
I agree with @junr03 that I'd prefer we stick with one approach (instance-based filters) rather than supporting static implementations (even if it's minimal work to add that support). Even if it's not something we bubble up to the public interface, it's still something we'd be implicitly supporting internally and subject to breakage, and I'm not sure I see much value in providing that functionality

Reply:
I'm also a little concerned that having this in place might lead us to miss actual problems with instance-based filters (i.e., `init_filter` being accidentally unset)

Reply:
Talked to @rebello95 and upon further consideration, I think the two of you make a good point and we should prefer simplicity. I'll switch this over to logging if init_filter isn't provided, and nulling out the rest of the struct for safety.